--!strict
local sha256 = require('./init')

local n = sha256.new():update('The quick brown fox jumps over the lazy dog') :: sha256.Sha256
print(n:hex())

-- // {
--     //     block = 0,
--     //     blocks = {
--     //         [0] = 32,
--     //         [1] = 99,
--     //         [2] = 114,
--     //         [3] = 32,
--     //         [4] = 32,
--     //         [5] = 112,
--     //         [6] = 118,
--     //         [7] = 116,
--     //         [8] = 108,
--     //         [9] = 32,
--     //         [10] = 26368,
--     //         [11] = 0,
--     //         [12] = 0,
--     //         [13] = 0,
--     //         [14] = 0,
--     //         [15] = 0,
--     //         [16] = 0,
--     //     },
--     //     bytes = 43,
--     //     finalized = false,
--     //     first = true,
--     //     h0 = 1779033703,
--     //     h1 = 3144134277,
--     //     h2 = 1013904242,
--     //     h3 = 2773480762,
--     //     h4 = 1359893119,
--     //     h5 = 2600822924,
--     //     h6 = 528734635,
--     //     h7 = 1541459225,
--     //     hBytes = 0,
--     //     hashed = false,
--     //     lastByteIndex = 43,
--     //     start = 43,
--     // }
    